src/libsyntax/parse/literal.rs

```diff
@@ -145,14 +148,19 @@ impl LitKind {
             }
             LitKind::Int(n, ty) => {
                 let suffix = match ty {
-                    ast::LitIntType::Unsigned(ty) => Some(Symbol::intern(ty.ty_to_string())),
-                    ast::LitIntType::Signed(ty) => Some(Symbol::intern(ty.ty_to_string())),
+                    ast::LitIntType::Unsigned(ty) => Some(ty.ty_to_symbol()),
+                    ast::LitIntType::Signed(ty) => Some(ty.ty_to_symbol()),
                     ast::LitIntType::Unsuffixed => None,
                 };
-                (token::Lit::Integer(Symbol::intern(&n.to_string())), suffix)
+                let n = if n < 10 {
+                    sym::integer(n as usize)
+                } else {
+                    Symbol::intern(&n.to_string())
```

Comment:
`sym::integer` can use signature `fn integer<N: TryInto<usize>>(n: N)` and fall back to interning automatically if the input doesn't fit into `usize`.

Reply:
Neat!